### ISSUE TITLE
Sorting broken?

### ISSUE THREAD

**Issue by jonas-w** [NeoApplications/Neo-Store#403]
I just searched for "pomodoro" and the sorting was completely wrong. 
I sort by time of update descending (newest first, right?).
The first one was last updated 15.12.2021.
The second one was last updated 07.06.2014.
The third one was last updated 28.08.2012
So far so good. But i wondered why there were no newer/more recently updated pomodoro apps.
Then i checked the fourth one and there it is 04.08.2022.
Then the fifth one 23.01.2022.
The sixth one 03.05.2022.
And the last one 27.12.2020.

The first three were ordered but the last few (which should come in front of the first three) were in no particular order.

![Screenshot_20230123_011244_Neo Store](https://user-images.githubusercontent.com/32615971/213948159-6f16dfb3-4d71-4c37-b8ff-77b7d4723889.png)

**Comment by machiav3lli** [comment]
Not really, when searching the first criteria of sorting is "how similar" the app to your search query. This is based on points system, which give an app more points if the search query is found in the app name, package name, short or long description. So maybe Goodtime includes "pomodore" in more places.

**Comment by jonas-w** [comment]
Makes sense if i don't want sorting.
But if i choose sorting, i want it to be sorted. 

There should be a neutral setting that does this algorithmic thing you mentioned, but the sort descending/ascending settings should sort your search result. Otherwise i don't see a point in a sorting option, if it doesn't really sort.

**Comment by machiav3lli** [comment]
I don't think you understood what I've said, but any way. I'll consider adding a "less-reliable sorted search mode"...

**Comment by Mrnofish** [comment]
No, what you said is clear enough nevertheless jonas raised a very solid point, IMO.

If you use e.g. update date descending to make sure you don't get old, likely abandoned apps that often do not work well or at all with newer versions of Android, you don't want an app from 2014 boosted up just because it fits the search key better than others.

The point based algorithm you mention does have value, however applying it to update date descending breaks with expectations predictability and this is in all likelihood the reason why this has been reported as a bug.

Perhaps you could have a "Smart Sort | Mode | Boost" that can be activated independently. E.g. a checkmark that the user can select or deselect to apply on top of the basic sorting strategy.

**Comment by walrus543** [comment]
Has there been no news regarding this issue for two years?
In my opinion, when searching with keywords, NeoStore should search both titles and descriptions. Furthermore, if I sort by *update date* in descending order, there is absolutely no reason for older apps to appear at the top of the list - which is exactly what happens currently.
I hope you will reconsider the *low priority* tag, as this creates a significant problem when trying to find new apps.
